### ISSUE TITLE
get_digraph removes only one edge among multiple parallel edges

### ISSUE THREAD

**Issue by nealeyoung** [gboeing/osmnx#807]
https://github.com/gboeing/osmnx/blob/677d13f56fa4894d51e07d502b1513f677262015/osmnx/utils_graph.py#L343

Consider the following code, for get_digraph.  I think it is incorrect.

 ```
   # identify all the parallel edges in the MultiDiGraph
    parallels = ((u, v) for u, v, k in G.edges(keys=True) if k > 0)

    # remove the parallel edge with greater "weight" attribute value
    for u, v in set(parallels):
        k, _ = max(G.get_edge_data(u, v).items(), key=lambda x: x[1][weight])
        to_remove.append((u, v, k))

    G.remove_edges_from(to_remove)
```

It apparently removes only *one* edge from each set of parallel edges.  This works in the case that the u-v pair has two parallel edges, but in the case that the u-v pair has three or more parallel edges, it won't --- after removing just one edge, the node pair (u, v) will still have  parallel edges.  I would have expected the code to instead delete all u-v edges except the one with minimum weight.

Am I missing something?

**Comment by gboeing** [comment]
I think I see your reasoning here. So, rather than removing only the max k value from the edges, we should remove all but the min k value from the edges?

**Comment by nealeyoung** [comment]
That makes sense to me, yes.

**Comment by gboeing** [comment]
Right. Would you like to open a PR to contribute this?

**Comment by nealeyoung** [comment]
I'm willing but don't know what that means.  I can suggest code if that would help.

**Comment by gboeing** [comment]
Sure. Proposed code to resolve this would be useful.

**Comment by nealeyoung** [comment]
Maybe replace the inner loop as follows?

```
  # identify all the parallel edges in the MultiDiGraph
   parallels = ((u, v) for u, v, k in G.edges(keys=True) if k > 0)

   # keep the parallel edge with minimum "weight" attribute value
   for u, v in set(parallels):
       k, _ = min(G.get_edge_data(u, v).items(), key=lambda x: x[1][weight])
       to_remove.extend((u, v, k1) for k1, _ in G.get_edge_data(u, v).items() if k1 != k)

   G.remove_edges_from(to_remove)
```

I haven't compiled or tested this code.

**Comment by nealeyoung** [comment]
I think there is another bug, namely, in the identification of parallel edges here:
```
   # identify all the parallel edges in the MultiDiGraph
   parallels = ((u, v) for u, v, k in G.edges(keys=True) if k > 0)
```

It seems to me that the condition checked (existence of (u, v, k) with k > 0) is neither necessary nor sufficient for the graph to have a parallel edge.  I think it's possible that (u, v) can have just one edge, but with k > 0 (e.g. after another edge has been deleted).  I'm not sure about this, but I think I am encountering this case in my code.

Here is a possible patch:

```
       parallels = ((u, v) for u, v in G.edges(keys=False) if len(G.get_edge_data(u, v)) > 0)
```

I am not a networkx expert though, so caveat emptor.

**Comment by gboeing** [comment]
@nealeyoung this should be fixed now via #826 if you'd be so kind as to take a look.
